x-pack/solutions/observability/plugins/streams_app/public/components/stream_detail_enrichment/processor_outcome_preview.tsx

```diff
@@ -0,0 +1,247 @@
+    isUpdate
+  ) => {
+    if (!isUpdate) {
+      return onTimeRangeRefresh();
+    }
+
+    if (dateRange) {
+      onTimeRangeChange({
+        from: dateRange.from,
+        to: dateRange?.to,
+        mode: dateRange.mode,
+      });
+    }
+  };
+
+  const getFilterButtonPropsFor = (filterId: DocsFilterOption) => ({
+    hasActiveFilters: docsFilter === filterId,
+    onClick: () => onDocsFilterChange(filterId),
+  });
+
+  return (
+    <EuiFlexGroup alignItems="center" justifyContent="spaceBetween" wrap>
+      <EuiFilterGroup
+        aria-label={i18n.translate(
+          'xpack.streams.streamDetailView.managementTab.enrichment.processorFlyout.outcomeControlsAriaLabel',
+          { defaultMessage: 'Filter for all, matching or unmatching previewed documents.' }
+        )}
+      >
+        <EuiFilterButton {...getFilterButtonPropsFor(docsFilterOptions.outcome_filter_all.id)}>
+          {docsFilterOptions.outcome_filter_all.label}
+        </EuiFilterButton>
+        <EuiFilterButton
+          {...getFilterButtonPropsFor(docsFilterOptions.outcome_filter_matched.id)}
+          badgeColor="success"
+          numActiveFilters={
+            simulationSuccessRate ? parseFloat((simulationSuccessRate * 100).toFixed(2)) : undefined
+          }
+        >
+          {docsFilterOptions.outcome_filter_matched.label}
+        </EuiFilterButton>
+        <EuiFilterButton
+          {...getFilterButtonPropsFor(docsFilterOptions.outcome_filter_unmatched.id)}
+          badgeColor="accent"
+          numActiveFilters={
+            simulationFailureRate ? parseFloat((simulationFailureRate * 100).toFixed(2)) : undefined
+          }
+        >
+          {docsFilterOptions.outcome_filter_unmatched.label}
+        </EuiFilterButton>
+      </EuiFilterGroup>
+      <StreamsAppSearchBar
+        onQuerySubmit={handleQuerySubmit}
+        onRefresh={onTimeRangeRefresh}
+        dateRangeFrom={timeRange.from}
+        dateRangeTo={timeRange.to}
+      />
+    </EuiFlexGroup>
+  );
+};
+
+interface OutcomePreviewTableProps {
+  documents: Array<Record<PropertyKey, unknown>>;
+  columns: string[];
+}
+
+const OutcomePreviewTable = ({ documents, columns }: OutcomePreviewTableProps) => {
+  if (isEmpty(documents)) {
+    return (
+      <EuiEmptyPrompt
+        titleSize="xs"
+        icon={<AssetImage type="noResults" />}
+        title={
+          <h2>
+            {i18n.translate(
+              'xpack.streams.streamDetailView.managementTab.rootStreamEmptyPrompt.noDataTitle',
+              { defaultMessage: 'Unable to generate a preview' }
+            )}
+          </h2>
+        }
+        body={
+          <p>
+            {i18n.translate(
+              'xpack.streams.streamDetailView.managementTab.enrichment.processorFlyout.outcomePreviewTable.noDataBody',
+              {
+                defaultMessage:
+                  'There are no samples document to test the processors. Try updating the time range or ingesting more data.',
```

Comment:
Question: is this wording completely accurate in the case that we do successfully fetch samples, but we don't necessarily get any matches from the processors?

Reply:
You are right, good catch! I reworded that a little to be more specific.